# Lung Cancer Prediction using Logistic Regression

This notebook builds a binary classification workflow using demographic and health-related survey features. The work includes basic preprocessing, label encoding, train/test splitting, model training, and performance evaluation using standard classification metrics.

**Main tasks covered**
- Encoding categorical variables
- Selecting input features
- Splitting data into training and testing sets
- Training a Logistic Regression model
- Evaluating with accuracy, confusion matrix, and classification report


## 1. Data Loading and Inspection


In [1]:
import pandas as pd
df = pd.read_csv('/content/survey lung cancer.csv')
df.head()

,GENDER,AGE,SMOKING,YELLOW_FINGERS,ANXIETY,PEER_PRESSURE,CHRONIC DISEASE,FATIGUE,ALLERGY,WHEEZING,ALCOHOL CONSUMING,COUGHING,SHORTNESS OF BREATH,SWALLOWING DIFFICULTY,CHEST PAIN,LUNG_CANCER
0,M,69,1,2,2,1,1,2,1,2,2,2,2,2,2,YES
1,M,74,2,1,1,1,2,2,2,1,1,1,2,2,2,YES
2,F,59,1,1,1,2,1,2,1,2,1,2,2,1,2,NO
3,M,63,2,2,2,1,1,1,1,1,2,1,1,2,2,NO
4,F,63,1,2,1,1,1,1,1,2,1,2,2,1,1,NO


In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 309 entries, 0 to 308
Data columns (total 16 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   GENDER                 309 non-null    object
 1   AGE                    309 non-null    int64 
 2   SMOKING                309 non-null    int64 
 3   YELLOW_FINGERS         309 non-null    int64 
 4   ANXIETY                309 non-null    int64 
 5   PEER_PRESSURE          309 non-null    int64 
 6   CHRONIC DISEASE        309 non-null    int64 
 7   FATIGUE                309 non-null    int64 
 8   ALLERGY                309 non-null    int64 
 9   WHEEZING               309 non-null    int64 
 10  ALCOHOL CONSUMING      309 non-null    int64 
 11  COUGHING               309 non-null    int64 
 12  SHORTNESS OF BREATH    309 non-null    int64 
 13  SWALLOWING DIFFICULTY  309 non-null    int64 
 14  CHEST PAIN             309 non-null    int64 
 15  LUNG_CANCER            

In [3]:
df['GENDER'] = df['GENDER'].map({'M': 0, 'F': 1}) #mapping male =0 , female =1
df['LUNG_CANCER'] = df['LUNG_CANCER'].map({'NO': 0, 'YES': 1}) #mapping cancer =1 , health = 0

In [4]:
df.head()

,GENDER,AGE,SMOKING,YELLOW_FINGERS,ANXIETY,PEER_PRESSURE,CHRONIC DISEASE,FATIGUE,ALLERGY,WHEEZING,ALCOHOL CONSUMING,COUGHING,SHORTNESS OF BREATH,SWALLOWING DIFFICULTY,CHEST PAIN,LUNG_CANCER
0,0,69,1,2,2,1,1,2,1,2,2,2,2,2,2,1
1,0,74,2,1,1,1,2,2,2,1,1,1,2,2,2,1
2,1,59,1,1,1,2,1,2,1,2,1,2,2,1,2,0
3,0,63,2,2,2,1,1,1,1,1,2,1,1,2,2,0
4,1,63,1,2,1,1,1,1,1,2,1,2,2,1,1,0


## 2. Model Setup and Feature Selection


In [5]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report

In [7]:
df.columns = df.columns.str.strip()
X = df[['GENDER','AGE','SMOKING','YELLOW_FINGERS','ANXIETY','PEER_PRESSURE','CHRONIC DISEASE','FATIGUE','ALLERGY','WHEEZING','ALCOHOL CONSUMING','COUGHING','SHORTNESS OF BREATH','SWALLOWING DIFFICULTY','CHEST PAIN']] #features columns where model learn from

y = df['LUNG_CANCER'] #target (perdiction)

## 3. Train/Test Split


In [8]:
#  Train-test split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [9]:
X_train

,GENDER,AGE,SMOKING,YELLOW_FINGERS,ANXIETY,PEER_PRESSURE,CHRONIC DISEASE,FATIGUE,ALLERGY,WHEEZING,ALCOHOL CONSUMING,COUGHING,SHORTNESS OF BREATH,SWALLOWING DIFFICULTY,CHEST PAIN
126,0,51,2,1,1,1,1,2,1,2,2,2,2,1,2
109,0,53,1,1,1,1,2,2,2,1,2,1,2,1,2
247,0,67,1,2,1,1,1,2,1,2,2,2,2,1,1
234,0,77,1,2,1,2,1,2,2,2,2,2,1,1,1
202,0,74,2,1,1,1,2,2,2,2,2,1,1,2,2
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
188,0,65,2,2,2,2,2,1,1,1,1,1,1,1,1
71,1,66,2,2,2,2,1,2,1,2,1,2,2,2,1
106,1,61,2,2,2,2,2,2,1,1,1,1,2,2,1
270,1,70,2,1,1,1,1,2,1,1,1,1,2,1,1


In [10]:
X_test

,GENDER,AGE,SMOKING,YELLOW_FINGERS,ANXIETY,PEER_PRESSURE,CHRONIC DISEASE,FATIGUE,ALLERGY,WHEEZING,ALCOHOL CONSUMING,COUGHING,SHORTNESS OF BREATH,SWALLOWING DIFFICULTY,CHEST PAIN
288,1,61,1,2,2,2,1,1,2,2,1,2,1,2,1
9,0,53,2,2,2,2,2,1,2,1,2,1,1,2,2
57,0,73,1,1,1,1,2,1,2,1,2,2,2,2,2
60,0,70,1,2,1,2,2,2,2,2,2,2,1,2,2
25,0,65,1,2,2,1,1,2,1,2,2,2,2,2,2
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
198,1,73,2,2,2,1,2,1,2,1,2,1,1,1,1
195,0,69,1,2,2,1,1,1,1,2,2,2,2,2,1
210,0,54,2,1,1,1,1,1,2,2,2,2,1,1,1
224,1,62,2,1,1,2,1,2,2,2,2,2,1,2,2


In [11]:
from  sklearn.linear_model import LogisticRegression

## 4. Model Training


In [12]:
#MODEL
model = LogisticRegression(max_iter=200000)
model.fit(X_train, y_train)

LogisticRegression(max_iter=200000)

## 5. Prediction and Evaluation


In [13]:
#  Make predictions
y_pred = model.predict(X_test)


In [14]:
y_pred

array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1])

In [15]:
from sklearn.metrics import accuracy_score

In [16]:
accuracy_score(y_test, y_pred)

0.967741935483871

In [17]:
print("✅ Accuracy:", accuracy_score(y_test, y_pred))
print("\n📊 Confusion Matrix:\n", confusion_matrix(y_test, y_pred))
print("\n📈 Classification Report:\n", classification_report(y_test, y_pred))

✅ Accuracy: 0.967741935483871

📊 Confusion Matrix:
 [[ 1  1]
 [ 1 59]]

📈 Classification Report:
               precision    recall  f1-score   support

           0       0.50      0.50      0.50         2
           1       0.98      0.98      0.98        60

    accuracy                           0.97        62
   macro avg       0.74      0.74      0.74        62
weighted avg       0.97      0.97      0.97        62



## 6. Summary

The notebook uses Logistic Regression to predict lung cancer outcomes from survey-based features.  
The saved result shows a **test accuracy of approximately 96.8%**, along with a confusion matrix and classification report for model assessment.
